src/main/java/uk/gov/companieshouse/api/accounts/exception/handler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.companieshouse.api.accounts.exception.handler;
+
+import static uk.gov.companieshouse.api.accounts.CompanyAccountsApplication.APPLICATION_NAME_SPACE;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import org.springframework.dao.DataAccessException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.NoHandlerFoundException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import uk.gov.companieshouse.api.accounts.exception.ApiException;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+  @ExceptionHandler(value = {DataAccessException.class, IOException.class,
+      IllegalArgumentException.class, IllegalStateException.class, NullPointerException.class,
+      RuntimeException.class, NoSuchAlgorithmException.class, Exception.class})
+  @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
+  protected void handleException(Exception ex) {
+    logError(ex, getExceptionMessage(ex));
+  }
+
+  @ExceptionHandler(value = {NoHandlerFoundException.class})
+  @ResponseStatus(value = HttpStatus.NOT_FOUND)
+  protected void handleNoHandlerFoundException(Exception ex) {
+    logError(ex, ExceptionMessage.NO_HANDLER_FOUND_EXCEPTION);
```

Comment:
Does Spring run these in order that they are defined? If so, couldn't this be simplified to:
```java
@ExceptionHandler(value = {NoHandlerFoundException.class})
@ResponseStatus(value = HttpStatus.NOT_FOUND)
protected void handleNotFoundException(Exception ex) {...}

@ExceptionHandler(value = {ApiException.class})
protected void handleApiException(ApiException apiex) {...}

@ExceptionHandler(value = {Exception.class})
@ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
protected void handleException(Exception ex) {..}
```